### ISSUE TITLE
JAWS doesn't navigate overflow menus correctly

### ISSUE THREAD

**Issue by manolyaa** [carbon-design-system/carbon#7445]
Here's a thread with an explanation of what I'm seeing and a demo https://ibm-analytics.slack.com/archives/C036P1CTN/p1607619522139900
When you open an overflow menu, JAWS announces that you’re at a menu and to use arrow keys to move through menu items, but when you hit the down arrow to move to the next menu item, JAWS says “leaving menu”. And when you keep pressing the down arrow to move through more items, focus doesn't move, and hitting enter on one of those menu items does not do the right thing.

## Environment

> Operating system
Running windows 10 in vmware

> Browser
Firefox

> Assistive technology used to verify
JAWS

This happens in the react storybook example https://react.carbondesignsystem.com/?path=/story/overflowmenu--basic

**Comment by tombrunet** [comment]
We ran into this also. Seems to be the worst when using Firefox. What I think is happening...
```
<li class="bx--overflow-menu-options__option" role="menuitem">
  <button class="bx--overflow-menu-options__btn" title="Copy an existing checklist" tabindex="-1" index="1">
    <div class="bx--overflow-menu-options__option-content">Copy an existing checklist</div>.
  </button>
</li>
```
So, it's a button inside a menuitem (though looks like it can also be a link in a menuitem depending on how you use it). I'm not surprised it's confused by that. It's reading it twice because there are two of them and it's leaving the menu because it thinks it's not a menu anymore. I expect to see something more like:
```
<li class="bx--overflow-menu-options__option" role="none">
  <button class="bx--overflow-menu-options__btn" title="Copy an existing checklist" tabindex="-1" index="1" role="menuitem">
    <div class="bx--overflow-menu-options__option-content">Copy an existing checklist</div>.
  </button>
</li>
```
This aligns with w3c guidance: https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-links.html where they make the `li[role=none]` and put `role=menuitem` on the link.

Created PR: https://github.com/carbon-design-system/carbon/pull/8599
